src/black/lines.py

```diff
@@ -165,6 +165,13 @@ def is_def(self) -> bool:
             and second_leaf.value == "def"
         )
 
+    @property
+    def is_stub_def(self) -> bool:
+        """Is this line a function definition with a body consisting only of "..."?"""
+        return self.is_def and self.leaves[-3:] == [
```

Comment:
Doesn't this also match something like `def f(x): y = ...`?

Reply:
Yes, this is true (and true for the is `is_stub_class` heuristic above as well). I've made the heuristic slightly better by including the colon. I didn't see anything adverse in the diff shades output.

Also note some of the pyi code we now use in linegen.py will trigger in some cases involving other statements. Didn't see anything in diff shades and it's not unreasonable when it happens.